nexus-repository-p2/src/main/java/org/sonatype/nexus/repository/p2/upgrade/AbstractP2Upgrade.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2017-present Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.p2.upgrade;
+
+import java.util.List;
+import java.util.Optional;
+
+import javax.inject.Provider;
+
+import org.sonatype.nexus.orient.DatabaseInstance;
+import org.sonatype.nexus.orient.DatabaseUpgradeSupport;
+import org.sonatype.nexus.orient.OClassNameBuilder;
+import org.sonatype.nexus.orient.OIndexNameBuilder;
+
+import com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx;
+import com.orientechnologies.orient.core.db.record.OIdentifiable;
+import com.orientechnologies.orient.core.index.OIndex;
+import com.orientechnologies.orient.core.metadata.schema.OSchemaProxy;
+import com.orientechnologies.orient.core.record.impl.ODocument;
+import com.orientechnologies.orient.core.sql.OCommandSQL;
+import com.orientechnologies.orient.core.sql.query.OSQLSynchQuery;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.util.stream.Collectors.toList;
+
+abstract class AbstractP2Upgrade
+    extends DatabaseUpgradeSupport
+{
+  private static final String SELECT_P2_REPOSITORIES = "select from repository where recipe_name = 'p2-proxy'";
+
+  private static final String P_REPOSITORY_NAME = "repository_name";
+
+  private static final String C_BROWSE_NODE = "browse_node";
+
+  private static final String BROWSE_NODE_CLASS = new OClassNameBuilder().type(C_BROWSE_NODE).build();
+
+  private static final String DELETE_BROWSE_NODE_FROM_REPOSITORIES =
+      String.format("delete from %s where repository_name in ?", BROWSE_NODE_CLASS);
+
+  private static final String I_REPOSITORY_NAME =
+      new OIndexNameBuilder().type("bucket").property(P_REPOSITORY_NAME).build();
+
+  private static final String ASSET_CLASS_NAME = "asset";
```

Comment:
the variable  not used

Reply:
> the variable not used

Fixed, I had intended to replace the vars with the same name in the subclasses.